osu.Game/Screens/Play/SoloSpectatorPlayer.cs

```diff
@@ -50,8 +48,7 @@ protected override void Dispose(bool isDisposing)
         {
             base.Dispose(isDisposing);
 
-            if (SpectatorClient != null)
-                SpectatorClient.OnUserBeganPlaying -= userBeganPlaying;
+            SpectatorClient.OnUserBeganPlaying -= userBeganPlaying;
```

Comment:
This is incorrect, `SpectatorClient` is resolved via DI, not guaranteed to be non-null in disposal. This is one minor pain point we have with the NRT system, but we get past that in the interim by using the `IsNotNull` extension:
```diff
diff --git a/osu.Game/Screens/Play/SoloSpectatorPlayer.cs b/osu.Game/Screens/Play/SoloSpectatorPlayer.cs
index 2cc1c4a368..be83a4c6b5 100644
--- a/osu.Game/Screens/Play/SoloSpectatorPlayer.cs
+++ b/osu.Game/Screens/Play/SoloSpectatorPlayer.cs
@@ -2,6 +2,7 @@
 // See the LICENCE file in the repository root for full licence text.
 
 using osu.Framework.Allocation;
+using osu.Framework.Extensions.ObjectExtensions;
 using osu.Framework.Screens;
 using osu.Game.Online.Spectator;
 using osu.Game.Scoring;
@@ -48,7 +49,8 @@ protected override void Dispose(bool isDisposing)
         {
             base.Dispose(isDisposing);
 
-            SpectatorClient.OnUserBeganPlaying -= userBeganPlaying;
+            if (SpectatorClient.IsNotNull())
+                SpectatorClient.OnUserBeganPlaying -= userBeganPlaying;
         }
     }
 }
```
(and well, it's kinda our mistake here, because we lie to the compiler that non-null DI'd properties are never null anywhere, so it's our responsibility to guard in places where it *may* be null)